### DISCUSSION TITLE
More icons for Workspaces. And prevent workspace from being deleted. Edit: Move workspaces and show workspace name when hovering with mouse.

### DISCUSSION THREAD

**Discussion by AKL55** [Floorp-Projects/Floorp#491] [Feature Request]
Similar to #346.

Would it be possible to have more icons available, or perhaps the option to add custom local icons? I have a Workspace named 'Music,' but unfortunately, there are no Music icons available at the moment.

I would use icons from this website here https://pictogrammers.com/

Adding a option to prevent Workspaces from being deleted if all tabs are  closed.

Edit:
Since there's no API to make tabs from workspaces undelete, I suggest adding a way to change workspace position in the sidebar. This way, 
if we delete the workspace, we can create a new one and move it to the same spot again.

Also, I forgot to add a feature request: to show the workspace's name when hovering with the mouse.

**Comment by CutterKnife** [comment]
Workspaces icons are created by me.
Thanks for the request. I will consider it.

**Reply by CutterKnife** [reply]
Thank you for your long patience. With the latest update, I have added 17 new icons, including the "Music" icon!

**Comment by Ryosuke-Asano** [comment]
> Similar to #346.
> 
> Would it be possible to have more icons available, or perhaps the option to add custom local icons? I have a Workspace named 'Music,' but unfortunately, there are no Music icons available at the moment.
> 
> I would use icons from this website here https://pictogrammers.com/
> 
> Adding a option to prevent Workspaces from being deleted if all tabs are closed.

It’s difficult to make workspaces undelete.

I can’t find api about after close tab event listener.

**Reply by AKL55** [reply]
I understand.
I edited the post with a workaround in case we delete the workspace accidentally.

**Reply by hannes3120** [reply]
Perhaps I missed a setting but something like "when last tab in a window is closed open a new tab instead of closing the window"-setting would allow for a decent workaround to prevent Workspaces from being deleted when the last tab is closed
